web/app.js

```diff
@@ -1329,12 +1329,33 @@ const PDFViewerApplication = {
       pdfViewer.optionalContentConfigPromise.then(optionalContentConfig => {
         this.pdfLayerViewer.render({ optionalContentConfig, pdfDocument });
       });
+      if ("requestIdleCallback" in window) {
+        window.requestIdleCallback(() => {
+          this._collectTelemetry();
+        });
+      }
     });
 
     this._initializePageLabels(pdfDocument);
     this._initializeMetadata(pdfDocument);
   },
 
+  /**
+   * A place to fetch data for telemetry after one page is rendered and the
+   * viewer is idle.
+   */
+  async _collectTelemetry() {
+    if (!this.pdfDocument) {
+      return;
+    }
+    const markInfo = await this.pdfDocument.getMarkInfo();
+    const tagged = markInfo ? markInfo.Marked : false;
```

Comment:
Nit: If you want to, you can now use optional chaining here[1] instead:
```suggestion
    const tagged = markInfo?.Marked || false;
```

---
[1] With the *exception* of the `src/core/` and `src/shared/` folders, you should now be free to use both optional changing and nullish coalescing where it makes sense.